client/components/track-interactions/index.js

```diff
@@ -16,29 +16,23 @@ const stringShape = PropTypes.oneOfType( [
 	PropTypes.arrayOf( PropTypes.string )
 ] );
 
-export default class TrackInteractions extends Component {
+class TrackInteractions extends Component {
 	static propTypes = {
 		children: PropTypes.element.isRequired,
 		mapPropsToAction: PropTypes.func,
 		fields: stringShape,
+
+		dispatch: PropTypes.func.isRequired,
 	}
 
 	static defaultProps = {
 		mapPropsToAction: constant( {} ),
 	}
 
-	static contextTypes = { store: storeShape };
-
-	constructor( props, context ) {
-		super( props, context );
-		const store = ( props.store || context.store );
-		this.dispatch = store ? store.dispatch : noop;
-	}
-
 	track = ( child ) => {
 		const action = this.mapPropsToAction( child.props );
 
-		this.dispatch( {
+		this.props.dispatch( {
 			type: COMPONENT_INTERACTION_TRACKED,
```

Comment:
Maybe put this inside an action creator. Import it and use `mapDispatchToProps` in connect. Looks like more standard way of doing this.